docs/packages/docs-utilities/package.json

```diff
@@ -0,0 +1,6 @@
+{
+  "name": "@mui-internal/docs-utilities",
+  "version": "1.0.0",
+  "private": "true",
+  "main": "index.ts"
```

Comment:
I'm not sure this setup will work well going forward. Every consumer of this package will have to transpile it. And I'm not sure how this would convert to an ESM package.

Reply:
It does work with the existing scripts (they are in TS), and we tend to write new ones in TS as well, so this shouldn't be a problem. If we hit any roadblock, we can rewrite it to JS + .d.ts.
As for ESM, TypeScript should handle it (it does respect the `type: module`), but I won't know for sure until I try.

Reply:
Ok, my main worry is that we're introducing a non-standard package layout here just (it seems to me) to avoid having to build it. Sooner or later one of these packages will need to be consumed by a target that isn't `ts-node` and we'll be back at the start again trying to consume non-js and non-standard package layouts.

Reply:
OK, that's a fair point. Assuming we stick to TS, how would you ensure that the build output is up to date when imported by a downstream package?

Reply:
I believe upstream packages should externalize their dependencies in the build. So that their build output only depends on code contained in the package itself. That way there doesn't need to be any synchronization between the builds of the different packages. At least that's how bundlers like e.g. `tsup` [do it](https://tsup.egoist.dev/#excluding-packages).

At the end of the line it's next.js that bundles all packages together and so you'd only need to wait with the production build of the docs until all the other packages have been built. This can also be automated with tooling like `turborepo`

Reply:
I didn't mean the production code. To give a concrete example - let's say we've got `@mui-internal/docs-utilities` written in TS and a utility JS script that uses it. How can we ensure that the `@mui-internal/docs-utilities` is transpiled to JS before it's used in the script? One way would be to check in the build output to git, but I'm not very keen on it.

Reply:
Ah yes, for that we'd need a monorepo task runner that takes topological order of the workspaces into account. e.g. you can specify for a command X in one package that it depends on a command Y in another package. That way you can specify a package that needs to be built before a command can run. It can be emulated in `lerna`  using their `--toposort` flag but that seems clunky. Tools like `turborepo` or `nx` are built for this use-case. 

Granted, probably not ideal to introduce such a big tooling decision in the current PR. But perhaps to consider as part of the North star for the infrastructure?

Reply:
Right, let's discuss the ideal shape of the infrastructure and the steps toward it elsewhere. For the time being, I'll change docs-utilities to JS + .d.ts. to keep the standards.